openfe_gromacs/protocols/gromacs_md/md_methods.py

```diff
@@ -0,0 +1,649 @@
+simulation using OpenMM tools and Gromacs.
+
+"""
+from __future__ import annotations
+
+import logging
+import pathlib
+import uuid
+from collections import defaultdict
+from collections.abc import Iterable
+from typing import Any
+
+import gufe
+import pint
+from gufe import ChemicalSystem, SmallMoleculeComponent, settings
+from openfe.protocols.openmm_utils import (
+    charge_generation,
+    settings_validation,
+    system_creation,
+    system_validation,
+)
+from openfe.protocols.openmm_utils.omm_settings import BasePartialChargeSettings
+from openfe.utils import log_system_probe, without_oechem_backend
+from openff.interchange import Interchange
+from openff.toolkit.topology import Molecule as OFFMolecule
+from openff.units import unit
+from openff.units.openmm import from_openmm, to_openmm
+from openmmtools import forces
+
+from openfe_gromacs.protocols.gromacs_md.md_settings import (
+    EMOutputSettings,
+    EMSimulationSettings,
+    GromacsMDProtocolSettings,
+    IntegratorSettings,
+    NPTOutputSettings,
+    NPTSimulationSettings,
+    NVTOutputSettings,
+    NVTSimulationSettings,
+    OpenFFPartialChargeSettings,
+    OpenMMEngineSettings,
+    OpenMMSolvationSettings,
+)
+
+logger = logging.getLogger(__name__)
+
+
+pre_defined_settings = {
+    "tinit": 0 * unit.picosecond,
+    "init_step": 0,
+    "simulation_part": 0,
+    "comm_mode": "Linear",
+    "nstcomm": 100,
+    "comm_grps": "system",
+    "verlet_buffer_tolerance": 0.005 * unit.kilojoule / (unit.mole * unit.picosecond),
+    "verlet_buffer_pressure_tolerance": 0.5 * unit.bar,
+    "coulomb_modifier": "Potential-shift",
+    "epsilon_r": 1,
+    "epsilon_rf": 0,
+    "fourierspacing": 0.12 * unit.nanometer,
+    "lj_pme_comb_rule": "Geometric",
+    "ewald_geometry": "3d",
+    "epsilon_surface": 0,
+    "nsttcouple": -1,
+    "tc_grps": "system",
+    "tau_t": 2.0 * unit.picosecond,
+    "pcoupltype": "isotropic",
+    "nstpcouple": -1,
+    "tau_p": 5 * unit.picosecond,
+    "compressibility": 4.5e-05 / unit.bar,
+    "lincs_warnangle": 30 * unit.degree,
+    "morse": "no",
+}
+
+pre_defined_settings_em = {
+    "emtol": 10.0 * unit.kilojoule / (unit.mole * unit.nanometer),
+    "emstep": 0.01 * unit.nanometer,
+}
+
+
+def _dict2mdp(settings_dict: dict, shared_basepath):
+    """
+    Write out a Gromacs .mdp file given a settings dictionary
+    :param settings_dict: dict
+          Dictionary of settings
+    :param shared_basepath: Pathlike
+          Where to save the .mdp files
+    """
+    filename = shared_basepath / settings_dict["mdp_file"]
+    # Remove non-mdp settings from the dictionary
+    settings_dict.pop("forcefield_cache")
+    settings_dict.pop("mdp_file")
+    with open(filename, "w") as f:
+        for key, value in settings_dict.items():
+            # First convert units to units in the mdp file, then remove units
+            if isinstance(value, pint.Quantity):
+                if value.is_compatible_with(unit.nanometer):
```

Comment:
With the upcoming to openff-models, this might go away - we can directly set it so that the settings object will always store the unit type that you want to use.

Reply:
Raised this as an issue